lib/identity_cache/cached/association.rb

```diff
@@ -39,6 +39,42 @@ def inverse_name
         end
       end
 
+      def fetch(records)
+        return if fetched?(records)
```

Comment:
`fetched?` runs the same checks as `prefetch_embedded` before calling `fetch_embedded` which is now below 👇